Document the new predefined edition behavior in tables

FabienLelaquais opened the pull request:
Addresses #1036.

FabienLelaquais committed:
Mention Polars support

jrobinAV reviewed (approved):
LGTM, for my part.

FabienLelaquais committed:
- Added doc for style parameter of Page classes
- Added dynamic styling example
- Release Notes update

FabienLelaquais committed:
Work in progress - focussing on the table control

FabienLelaquais committed:
Comments and images on doc examples

FabienLelaquais committed:
Add and update docs for table and selector

namnguyen20999 reviewed (approved):
lgtm

FredLL-Avaiga commented on docs/refmans/gui/viselements/generic/selector.md_template:
```diff
@@ -118,18 +210,95 @@ users = [
+
+The selector control list is generated as a collection of items grouped in an element that is
+assigned the "MuiList-root" class. The descendants (the list items) have the
+"MuiListItemButton-root" class applied.
```
link to Mui (one day) ?

FabienLelaquais replied:
Yeah... I did not do it here because of the multiple components at play.
It does make sense. That's what I've done in 'progress'...

FredLL-Avaiga commented on docs/release-notes/index.md:
```diff
@@ -35,10 +35,31 @@ This is the list of changes to Taipy releases as they were published.
+  provide a compact representation of a process's progress.
+- A new control, [`chat`](../refmans/gui/viselements/generic/chat.md), has been introduced to
+  simplify the development of chat-based applications.
+- The [`table`](../refmans/gui/viselements/generic/table.md) control has new features:
```
mention renaming style to row_class_name & cell_class_name[] ?

FabienLelaquais replied:
YES!
That's what I was forgetting.
Thanks!

FredLL-Avaiga commented on docs/release-notes/index.md:
```diff
@@ -65,24 +86,38 @@ This is the list of changes to Taipy releases as they were published.
   See [issue #1600](https://github.com/Avaiga/taipy/issues/1600).
+- The *decimator* property of the [`chart`](../refmans/gui/viselements/generic/chart.md) control
+  now applies to traces that are only "lines" or "markers".
+- A new toggle button has been added to the [`login`](../refmans/gui/viselements/generic/login.md)
```
same thing for input type password

FabienLelaquais replied:
Yeah... that would be part of the input doc though. The reference to the input type is done...

FredLL-Avaiga commented on docs/release-notes/index.md:
```diff
@@ -65,24 +86,38 @@ This is the list of changes to Taipy releases as they were published.
+  control, enabling users to show or hide the entered password.
+- Page Builder API:
+    - Properties that expect a function can now be set to a lambda function in the Page Builder
+      API.<br/>
```
properties that do not expect a function can be defined as lambda too

FabienLelaquais replied:
Done. Hopefully.

FredLL-Avaiga commented on docs/release-notes/index.md:
```diff
@@ -118,11 +153,19 @@ This is the list of changes to Taipy releases as they were published.
+  [this](../userman/gui/pages/builder.md#setting-property-values) section.<br/>
+  See [issue #1715](https://github.com/Avaiga/taipy/issues/1715).
+- Binding to an element of a collection is now supported.<br/>
+  See [issue #1785](https://github.com/Avaiga/taipy/issues/1785).
```
LoV as Enum

FabienLelaquais replied:
That was for... after. On it anyway.

FabienLelaquais replied:
Done.

FabienLelaquais committed:
Fred comments:
  - Added note on Lambdas as property values
  - Enumerated classes as candidates for *lov*

FredLL-Avaiga commented on docs/userman/gui/pages/builder.md:
````diff
@@ -126,6 +130,29 @@ Compared to the previous example, you can see that the label uses a bold font we
+    list. A lambda function makes this straightforward:<br/>
+    ```python
+    for name in ages:
+        tgb.text(lambda ages: f"{name} is {ages.get(name)}")
````
This example is a bit misleading
Because we use two versions of the ages variable: 
The frame variable in the for
The state variable in the parameter of the lambda
So even if an entry is added to the state variable, the result will not change
But it will change if you change one value if the dict 

I would propose
```
prefix = "prefix: "
for name, age in ages.entries():
  tgb.text(lambda prefix: f"{prefix}{name} is {age}"
```

And show that you can change the prefix state variable

FabienLelaquais replied:
That is SO true.
What about
```
import taipy.gui.builder as tgb
from taipy.gui import Gui

people = {
    "Albert": 1982,
    "Beatrix": 1955,
    "Cecilia": 2003
}
         
current_year = 2024

with tgb.Page() as page:
    tgb.text("Year: ", inline=True)
    tgb.slider("{current_year}", min=1990, max=2050)

    for name, birth_year in people.items():
        tgb.text(lambda current_year: f"{name} would be {current_year-birth_year}")

if __name__ == "__main__":
    Gui(page).run()
```

FredLL-Avaiga replied:
Excellent

FabienLelaquais committed:
Added doc on "Lambdas as property values" for the Page Builder API
Added doc on lovs from enumeration classes,
